bigframes/core/block_transforms.py

```diff
@@ -500,3 +500,68 @@ def _kurt_from_moments_and_count(
         kurt_id, na_cond_id, ops.partial_arg3(ops.where_op, None)
     )
     return block, kurt_id
+
+
+def align(
+    left_block: blocks.Block,
+    right_block: blocks.Block,
+    join: str = "outer",
+    axis: typing.Union[str, int, None] = None,
+) -> typing.Tuple[blocks.Block, blocks.Block]:
+    axis_n = core.utils.get_axis_number(axis) if axis is not None else None
+    if (axis_n == 0) or (axis_n is None):
```

Comment:
I would really encourage to rearrange and split the body into several separate helper parts, to make the logic more obvious. For example, it is hard to get the idea of "sync_blocks = True", and when it will apply. We can create a function with param "sync_blocks", call with True in axis == None, with False otherwise.

Reply:
Simplified by splitting into align_rows and align_columns and making sure that columns are aligned first